### DISCUSSION TITLE
Configuration based routing

### DISCUSSION THREAD

**Discussion by yan5** [sveltejs/kit#10327] [Ideas]
We (working with a fortune 100) have started using SvelteKit for a project and the experience so far has been very good, but as the project growing big, we're finding that those +page.. files are coming out as kind of an eye sore, esp since many people have started working on this project. Although we tried hard to adjust (setting label format to 'medium' in Vscode also not helping to be honest) but its just keep getting in the way of otherwise an amazing developer experience. 

I'm not sure if this has been discussed (and dismissed), but can we explore to have a routing based on config file, something similar to what Django (https://www.django-rest-framework.org/api-guide/routers/) or https://javalin.io/documentation#endpoint-handlers does.

Having a configuration to map 'routes + methods' to the corresponding 'file/modules/functions' would give flexibility for developers to chose files/modules at their discretion.

The syntax & semantics could be completely different, but the gist is something as follows

`addRoute('/foo/bar', {'page':'/site/bar.js', .....});
`

This could be completely optional to the exiting directory based routing.

Or am I missing something?

**Comment by maxcoredev** [comment]
I hope it's legal to share here my solution:
https://dev.to/maxcore/sveltekit-2-code-based-router-instead-of-file-based-router-now-is-just-a-gist-28cm
